algorithmic_efficiency/workloads/librispeech_conformer/librispeech_pytorch/workload.py

```diff
@@ -266,15 +266,18 @@ def _eval_model_on_split(self,
       decoded, decoded_paddings = self.greedy_decode(logits, logits_padding)
       targets, target_paddings = batch['targets']
       word_errors, num_words = metrics.compute_wer(
-          decoded=decoded.detach().cpu().numpy(),
-          decoded_paddings=decoded_paddings.detach().cpu().numpy(),
-          targets=targets.detach().cpu().numpy(),
-          target_paddings=target_paddings.detach().cpu().numpy(),
+          decoded=decoded.cpu().numpy(),
+          decoded_paddings=decoded_paddings.cpu().numpy(),
+          targets=targets.cpu().numpy(),
+          target_paddings=target_paddings.cpu().numpy(),
           tokenizer=self.tokenizer)
-      loss = self._loss_fn((targets, target_paddings), (logits, logits_padding))
+      mean_loss, per_example_losses = self.loss_fn(
+          (targets, target_paddings), (logits, logits_padding))
+      summed_loss = per_example_losses.sum()
+      lengths = summed_loss / mean_loss
```

Comment:
While this is correct theoretically, we might need to do a round operation to ensure that the `lengths` are whole numbers? In the eval step, we `all_reduce` the lengths and then use it for computing globally normalized ctc loss and am not totally sure if this would distort the loss-values in some way and by how much -- at this point, im okay either way :)

Reply:
Yeah, it doesn't hurt to round here. I'll fix it, thanks!